include/CL/sycl/parallelism/detail/parallelism_tbb.hpp

```diff
@@ -0,0 +1,201 @@
+#ifndef TRISYCL_SYCL_PARALLELISM_DETAIL_PARALLELISM_HPP
+#define TRISYCL_SYCL_PARALLELISM_DETAIL_PARALLELISM_HPP
+
+/** \file
+
+    Implement the detail of the parallel constructions to launch kernels
+
+    \todo Refactor this file
+
+    Ronan at keryell dot FR
+
+    This file is distributed under the University of Illinois Open Source
+    License. See LICENSE.TXT for details.
+*/
+
+#include <boost/multi_array.hpp>
+#include <cstddef>
+
+#include "CL/sycl/group.hpp"
+#include "CL/sycl/h_item.hpp"
+#include "CL/sycl/id.hpp"
+#include "CL/sycl/item.hpp"
+#include "CL/sycl/nd_item.hpp"
+#include "CL/sycl/nd_range.hpp"
+#include "CL/sycl/range.hpp"
+
+#include <tbb/blocked_range2d.h>
+#include <tbb/blocked_range3d.h>
+#include <tbb/parallel_for.h>
+
+/** \addtogroup parallelism
+    @{
+*/
+
+namespace cl {
+  namespace sycl {
+    namespace detail {
```

Comment:
`namespace cl::sycl::detail {` for new use cases since we are in C++17

Reply:
I totally missed that addition to C++17! Now I'll have to use that in all my C++17 projects. :)

Reply:
And C++20 is coming... Even better! :-)